sqlparser/ast.go

```diff
@@ -1670,9 +1696,9 @@ type ColIdent struct {
 	// This artifact prevents this struct from being compared
 	// with itself. It consumes no space as long as it's not the
 	// last field in the struct.
-	_              [0]struct{ _ []byte }
-	val, lowered   string
-	wrapWithQuotes bool
+	_            [0]struct{ _ []byte }
+	val, lowered string
+	quote        byte
```

Comment:
golang string type doesn't allow (https://play.golang.org/p/l1TOWaeOxGt) to assign `nil`, only `""` empty string. So we will need to use "special value" empty string. Plus string allows assigning more then 1 symbol what is incorrect in this context. We can use separate type as alias for byte... But don't sure that it needs here and will safe us from some unexpected errors. Imho variable name explains enough about value and declaring `quote quote` as struct field is unnecessarily

Reply:
(That's just my Unicode paranoid self speaking. If quote characters were not ASCII then we may have needed more that 1 byte for them. But we're safe as SQL dialects use only ASCII quotes.)

Using empty string may not be that bad... If we use empty string to represent "no quoting necessary" then we could always 'quote' identifiers by doing `quote + identifier + quote`, without making a check whether the quoting is necessary. However, this would be more like a 'normalization' operation rather than quoting.

Another thing that crossed my mind is to add utility methods like
```go
func (node TableIdent) IsQuoted() bool {
        return node.quote != 0
}
```

Anyhow, I'm fine with the current code. It's pretty much obvious where we mean 'no quoting' and it should not cause issues in practice.